src/providers/twitch/TwitchMessageBuilder.cpp

```diff
@@ -466,10 +467,16 @@ void TwitchMessageBuilder::appendChannelName()
 
 void TwitchMessageBuilder::parseUsername()
 {
-    auto iterator = this->tags.find("color");
-    if (iterator != this->tags.end())
+    const auto iterator = this->tags.find("color");
+    if (const auto color = iterator.value().toString(); !color.isEmpty())
+    {
+        this->usernameColor_ = QColor(color);
+    }
+    else if (getSettings()->colorizeNicknames &&
+        this->tags.contains("user-id"))
     {
-        this->usernameColor_ = QColor(iterator.value().toString());
+        const auto id = this->tags.value("user-id").toInt();
```

Comment:
User IDs should not be stored as ints, Twitch holds no guarantees about user IDs being convertible to numbers.